app/user-orders-repository.js

```diff
@@ -0,0 +1,34 @@
+const getUsersNames = async (db) => {
+    return new Promise((resolve, reject) => {
+        db.all(`SELECT DISTINCT name FROM USERS`, (err, rows) => {
+            if (err) {
+                reject(err);
+            } else {
+                resolve(rows.map(row => row.name));
+            }
+        });
+    });
+};
+
+const getUserTotal = async (db, userName) => {
```

Comment:
The code you highlighted constructs an SQL query by directly interpolating the userName variable into the query string. This approach is vulnerable to SQL injection attacks if userName is not properly sanitized.

To make it more secure, consider using parameterized queries or prepared statements to prevent SQL injection. Here is an example of how you can modify the code to use parameterized queries:

const getUserTotal = async (db, userName) => {
    return new Promise((resolve, reject) => {
        const sql = `SELECT SUM(total) as total FROM ORDERS o INNER JOIN USERS u ON u.id = o.user_id WHERE u.name = ?`;
        db.all(sql, [userName], (err, rows) => {
            if (err) {
                reject(err);
            } else {
                resolve({ name: userName, total: rows[0].total });
            }
        });
    });
};
This change will help mitigate the risk of SQL injection.